File: LeetCode_ProblemSet/Problem3375.cpp
#include <unordered_set>
#include <vector>

class Solution {
   public:
    int minOperations(std::vector<int>& nums, int k) {
        std::unordered_set<int> set;
        for (int num : nums) {
            if (num < k) return -1;
            if (num > k) set.insert(num);
        }
        return set.size();
    }
};

File: LeetCode_ProblemSet/Rust/src/Problem3375.rs
use std::collections::HashSet;

impl Solution {
    pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
        let mut set = HashSet::new();
        for num in nums {
            if num < k {
                return -1;
            }
            if num > k {
                set.insert(num);
            }
        }
        return set.len() as i32;
    }
}